ribs/archives/_dns_archive.py

```diff
@@ -456,11 +378,23 @@ def add(
 
         # Update stats.
         if len(self) > 0:
```

Comment:
Is there any case where adding will result in the archive shrinking in size? If so, this if statement would cause the archive stats to no longer be current.